### PR TITLE
Branches, Kristy

### PR DESCRIPTION
**Pull request by kristyh32**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The model is representing a single instance of an object from the database. It is also handling any business logic.
Describe in your own words what the Controller is doing in Rails | the controller talks to the Model and the View in order to decide what to give back to the user.
Describe in your own words what the View is doing in Rails | The view has different html files that will be loaded depending on where the router tells the controller to look.
Describe an edge-case controller test you wrote | An edge case test I wrote was trying to edit a task that was not found and expecting that it will redirect the user to the root page and give a flash error.
What is the purpose of using strong params? (i.e. the params method in the controller) | strong params allow us to specify what attributes we want an object to have and adds a layer of security so that users cannot modify things that we don't want them to.
How are Rails migrations related to Rails models? | migrations can make changes to the database schema and since models represent one instance of something from the database, the model will have to be updated to reflect any changes made to the db by the migrations.
Describe one area of Rails that are still unclear on | I am unclear on how to apply different css stylesheets for different views in the program.

### CODE REVIEW

**Comment by tildeee** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | x
Answered comprehension questions | x
Successfully handles: Index, Show | x
Index & Show tests pass | x
Successfully handles: New, Create | x
New & Create tests pass | x
Successfully handles: Edit, Update | x
Tests for Edit & Update test valid & invalid task ids | x
Successfully handles: Destroy, Task Complete | x
Tests for Destroy & Task Complete include tests for valid and invalid task ids | x
Routes follow RESTful conventions | x
Uses named routes (like `_path`) | x
**Overall** |



Well done on this project, Kristy! The app code and test code look great-- clean and functioning perfectly!


I've added only a few comments. That being said, this project was well done... keep up the good work!

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  def index
+    @tasks = Task.all
+    
+  end
+  
+  def show
+    task = params[:id]
+    @task = Task.find_by(id: task)
+    if @task.nil?
+      flash[:error] = "Could not find task with id: #{task}"
```
Comment (tildeee) [inline app/controllers/tasks_controller.rb:12]: You probably weren't able to use `flash` meaningfully-- we will learn about `flash` so very very soon!

File: test/controllers/tasks_controller_test.rb [hunk 2]
```diff
@@ -71,56 +69,133 @@
...
-  end
-
-  # Complete for Wave 4
-  describe "toggle_complete" do
-    # Your tests go here
-  end
-end
+    
+    # Complete for Wave 4
+    describe "toggle_complete" do
```
Comment (tildeee) [inline test/controllers/tasks_controller_test.rb:166]: Fantastic tests for the `toggle_complete` functionality! The last remaining edge case is trying to do this with an invalid id.
